src/fetch.rs

```diff
@@ -1,32 +1,17 @@
-use serde::{Deserialize, Serialize};
+use anyhow::{anyhow, Result};
+use cmsis_pack::{pack_index::Vidx, utils::FromElem};
 
-#[allow(non_snake_case)]
-#[derive(Serialize, Deserialize, Debug)]
-pub(crate) struct Request {
-    pub(crate) Values: Vec<Pack>,
-}
-
-#[allow(non_snake_case)]
-#[derive(Serialize, Deserialize, Debug)]
-pub(crate) struct Pack {
-    pub(crate) Id: String,
-    pub(crate) Name: String,
-    pub(crate) ShortName: String,
-    pub(crate) Vendor: String,
-    pub(crate) Description: String,
-    pub(crate) IsDevicePack: bool,
-    pub(crate) IsBoardPack: bool,
-    pub(crate) IsThirdParty: bool,
-    pub(crate) PackUrl: String,
-}
-
-pub(crate) fn list_packs() -> Result<Vec<Pack>, reqwest::Error> {
-    let packs = reqwest::blocking::Client::new()
-        .get("https://api.arm.com/e-cmsis/v3/packs")
-        .header("uuid", "97822e53-79f4-4ca0-a6cb-c8e0acb57282")
+/// Fetches the master VIDX/PIDX file from the ARM server and returns the parsed file.
+pub(crate) fn get_vidx() -> Result<Vidx> {
+    let reader = reqwest::blocking::Client::new()
+        .get("https://www.keil.com/pack/index.pidx")
         .send()?
-        .json::<Request>()?
-        .Values;
+        .text()?;
+
+    let vidx = match Vidx::from_string(&reader) {
+        Ok(vidx) => vidx,
+        Err(e) => return Err(anyhow!(e.to_string())),
```

Comment:
You could use `e.compat()`, then it should be possible to just the question mark to handle the result.